packages/api/src/controllers/baseController.js

```diff
@@ -195,7 +195,8 @@ export default {
     return await model
       .query(trx)
       .context({ user_id: req?.auth?.user_id, ...context })
-      .insertGraph(removeAdditionalPropertiesWithRelations(model, data));
+      .insertGraph(removeAdditionalPropertiesWithRelations(model, data))
+      .returning('*');
```

Comment:
Oh! I'm surprised we never had to do this before to get the auto-incrementing integers back.

Reply:
To be honest, I have no idea why `id` is returned...😅

Reply:
Sorry I'm not following! Why do we need all the properties returned here, is this mainly for tests?

Reply:
I don't think we need all the properties, but we at least need `internal_identifier` which wouldn't be returned without  `.returning('*')` to update the list of batches that we already have on the frontend.
If there's a way to not include unnecessary properties like created_by..., updated_by... etc. maybe we can go with that, but I'm not sure if we are doing it already or we should start doing it...